GetStream/stream-chat-swift
# [CIS-1264] Handle shadow bans

### 🔗 Issue Link

CIS-1264

### 🎯 Goal

Shadow ban means the banner user's messages will have `shadowed: true` flag in them and it's the Client's responsibility to not render those messages.
This flag will only be visible to users other than the shadow banned user so they have no way to realize they're shadow banned - ie, a shadow banned user won't see `shadowed: true` for their own messages.

### 🛠 Implementation

iOS SDK did not parse the `shadowed` flag on message, let alone use it. This PR handles it the same as `deletedMessagesVisibility`, by giving the user a choice on the matter. The user can either exclude these messages from the controller's fetch request (therefore not showing them) or include the messages (and render them differently, by providing a `ChatMessageContentView` subclass)

### 🧪 Testing

1. Shadow ban a user
2. Send a message from that user
3. Observe the same channel from the DemoApp

### 🎨 Changes

<img width="458" alt="Screen Shot 2021-11-01 at 17 48 02" src="https://user-images.githubusercontent.com/770464/139708853-53712bc8-1389-4cd0-a58e-ece30a22dd91.png">

### ☑️ Checklist
